submissions/OlexiyDobroskok/OOP_exercise/index.js

```diff
@@ -0,0 +1,159 @@
+const inhabitantDefaultProperties = {
+  species: "No species",
+  name: "No name",
+  gender: "No gender",
+  saying: "Hello",
+  friends: "",
+};
+
+const humanDefaultProperties = {
+  hands: 2,
+  legs: 2,
+};
+
+const petsDefaultProperties = {
+  paws: 4,
+};
+
+class Inhabitant {
+  static worldPopulation = [];
+  constructor({
+    species,
+    name,
+    gender,
+    saying,
+    friends,
+  } = inhabitantDefaultProperties) {
+    this.species = species ?? inhabitantDefaultProperties.species;
+    this.name = name ?? inhabitantDefaultProperties.name;
+    this.gender = gender ?? inhabitantDefaultProperties.gender;
+    this.saying = saying ?? inhabitantDefaultProperties.saying;
+    this.friends = friends ?? inhabitantDefaultProperties.friends;
+    Inhabitant.worldPopulation.push(this);
```

Comment:
Not a good idea to populate world from within inhabitants.
There is a Factory pattern in SW design, but this is not the case.

Reply:
Can I create a separate function that will create and push objects into an array?

Reply:
> Not a good idea to populate world from within inhabitants. There is a Factory pattern in SW design, but this is not the case.

Can I create a separate function that will create and push objects into an array?

Reply:
Yeah. Creating `World` class would be an overkill at this stage of project development.
But please avoid `.push`. `Array` methods provide more semantic ways to solve this.
And you will probably want to help inhabitants make friends after they are born. At least exisiting inhabitants cannot make friends with those who are not born yet.

Reply:
I abandoned this idea because it was originally intended to simplify the creation of the world, but Later there are problems with withdrawal of friends.What method can you recommend to replace the push?It seemed to me very convenient to work initially with an empty array. (

Reply:
I reviewed this fragment. I agree that `.push` is a way to go.
The only thing remains: it is a world that consists of Inhabitants rather than world is a part of any Inhabitants.
Sure, syntax allows us to do this and make a static prop looks attractive. But having world as a part of inhabitant is counter-intuitive.
World can be a collection of inhabitants. Simplest collection is JS array. If we wanted some world-wide operations we would have multiple functions for those operations and it would make sense to scope them under class World.